### DISCUSSION TITLE
Generate Access Token for non-Microsoft IDP and and call a downstream api

### DISCUSSION THREAD

**Discussion by cwistedt** [AzureAD/microsoft-identity-web#2667] [Q&A]
Hi,

I have tried to understand if i can use microsoft.identity.web for my scenario but is struggling.

Scenario:
* I have an Azure Function
* I need to generate a token for a 3 scale api (client_credential). The token endpoint is something like this:
  * https://{comain}/auth/realms/{realm}/protocol/openid-connect/token  
* I do not need top protect my azure function by authentication or authorization
* I need to call said api with the generated access_token.

Is this possible by microsoft.identity.web or should i use a different approach.

**Comment by jmprieur** [comment]
@cwistedt : I would not expect these IdP to work for the moment. Does the openid connect document express the token endpoint?
@bgavrilMS : any idea on how we could make it work?
